src/librustc_span/lib.rs

```diff
@@ -1063,18 +1063,87 @@ impl SourceFileHash {
     }
 }
 
-/// A single source in the `SourceMap`.
-#[derive(Clone)]
-pub struct SourceFile {
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SourceFileName {
     /// The name of the file that the source came from. Source that doesn't
     /// originate from files has names between angle brackets by convention
     /// (e.g., `<anon>`).
-    pub name: FileName,
-    /// `true` if the `name` field above has been modified by `--remap-path-prefix`.
-    pub name_was_remapped: bool,
+    name: FileName,
+    /// `true` if the name field was modified by `--remap-path-prefix`.
+    was_remapped: bool,
```

Comment:
Is this field necessary? Could it instead be derived from the state of `unmapped_name` and `name`?